integration/container_event_test.go

```diff
@@ -0,0 +1,133 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package integration
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"testing"
+	"time"
+
+	"github.com/containerd/containerd/integration/images"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	runtime "k8s.io/cri-api/pkg/apis/runtime/v1"
+)
+
+func TestContainerEvents(t *testing.T) {
+	ctx := context.Background()
+	defer ctx.Done()
+
+	t.Log("Set up container events streaming client")
+	containerEventsStreamingClient, err := runtimeService.GetContainerEvents(ctx, &runtime.GetEventsRequest{})
+	assert.NoError(t, err)
+	containerEventsChan := make(chan *runtime.ContainerEventResponse)
+
+	go listenToEventChannel(ctx, t, containerEventsChan, containerEventsStreamingClient)
+	// Sleep to make sure the event channel is set up, and then drain all events
+	// emitted by previous tests.
+	time.Sleep(2 * time.Second)
+	drainContainerEventsChan(t, containerEventsChan)
```

Comment:
With the `time.Sleep` here and the `time.After` in `drainContainerEventsChan`, we're waiting 2.5 seconds.  Is it necessary to wait that long?

Reply:
Good catch. Only the 500ms is needed so I removed the 2s sleep here.